### PR TITLE
Create UI for add-maintainer

### PR DESCRIPTION
**Pull request by jnnybk**
(NOTE) This PR should rebase master once PR #44 is merged.

This PR accomplishes:
1) Adding "Add Maintainer" option to the NavBar for Maintainers.
2) Creates Pop-Up to input User Email to send to /add-maintainer servlet.
Screenshot 2020-07-15 at 2 47 46 PM - Display 2

### CODE REVIEW

**Review by Sarah-Addo — COMMENTED** [review]
could you have the popup overlay the page like https://screenshot.googleplex.com/JJNuVhuxt6C?

**Comment by jnnybk** [comment]
Updates to this PR's completed tasks:
1) Created a modal for adding maintainer. 
2) Fix logic error in AddMaintainerServlet.java
3) Removed maintainer.js, added it to user.js when building the navBar

Screenshot 2020-07-15 at 7 30 26 PM - Display 2

**Review by andrewm898 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/AddMaintainerServlet.java [hunk 1]
```diff
@@ -53,13 +53,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
-    Entity entity = GivrUser.getUserFromDatastoreWithProperty("userEmail", currUser.getUserEmail());
+    Entity entity = GivrUser.getUserFromDatastoreWithProperty("userEmail", newMaintainerEmail);
     if (entity != null) {
-      changeMaintainerStatus(newMaintainerEmail, currUser.getUserId());
+      changeMaintainerStatus(newMaintainerEmail);
     } else {
       addNewMaintainerToDatastore(newMaintainerEmail);
     }
     response.setStatus(HttpServletResponse.SC_OK);
```
Comment (belraquib) [inline src/main/java/com/google/sps/servlets/AddMaintainerServlet.java]: You don't want to set OK and sendRedirect. Pick one.

File: src/main/webapp/user.js [hunk 2]
```diff
@@ -58,11 +91,30 @@ class User {
...
+      const addMaintainerLabel = document.createElement("a");
+      addMaintainerLabel.textContent = "Add Maintainer";
+      addMaintainerLabel.setAttribute("id", "add-maintainer-label");
+      addMaintainerLabel.addEventListener("click", () => {
+        const addMaintainerPopupBackground = document.createElement("div");
+        addMaintainerPopupBackground.setAttribute("id", "modal-popup-background");
+        addMaintainerPopupBackground.classList.add("add-maintainer-popup-modal-background");
+        addMaintainerPopupBackground.appendChild(addMaintainerPopup);
+
+        const mainContent = document.getElementById("help-near-me-content") ? document.getElementById("help-near-me-content") : document.getElementById("organizations-content");
```
Comment (belraquib) [inline src/main/webapp/user.js]: This seems weird to me. Why add it to the mainContent and not document.body ? As is, you are leaking what page is hosting this popup into the logic here. What happens when/if you have a new page?

**Review by belraquib — CHANGES_REQUESTED** [review]

File: src/main/webapp/user.js (hunk 2 above)
Comment (jnnybk) [reply src/main/webapp/user.js]: I agree! I fixed it to prepend to the body of the document instead of to specific pages.

**Review by belraquib — APPROVED** [review]

**Pull request merged**